html.cfg

```diff
@@ -29,10 +29,12 @@
     overflow-x: auto;
 }}
 
-\Css{.ecrm-0500 { 
+\Css{.ecrm-0500 {
     font-size: 70\%;
-    font-style: italic; 
+    font-style: italic;
     color: gray;
+    width: 1.5rem;
+    display: inline-block;
```

Comment:
I have tried your script, it seemed to be welled.
however there are several problems need to be fixed.
* The indention of makefile block is missing :
* line number in makefile block don't share the same horizontal line with codes :
![圖片](https://user-images.githubusercontent.com/58484289/128396681-154a2daf-e5f6-4693-8625-930ebf199e84.png)
(This can be seen in section `0.4.1`)
* some long codes are forced to break into a new line :
![圖片](https://user-images.githubusercontent.com/58484289/128397361-68dfcbd7-7e2a-4e2e-80de-1170ec35a6ba.png)
(This can be seen in section `0.6.1`)
* line number in code block don't share the same horizontal line with codes :
![圖片](https://user-images.githubusercontent.com/58484289/128397637-c48ab698-4ce5-4b33-a604-320bed65ef77.png)
(This can be seen in section `0.7`)

Reply:
Line number in `makefile` block is rendering well now:

![image](https://user-images.githubusercontent.com/26391143/128448168-3d07356e-f7e7-486e-9a82-9732b75efd67.png)

Long codes will be no-wrap and in the same line:

![image](https://user-images.githubusercontent.com/26391143/128448235-defe1d48-17a3-4411-b703-d2635a224393.png)

---

**But the indentation problem you show in Section 0.7 should fix in `*.c` source code. Please use spaces for indentation instead of tabs.**

Reply:
Long code still be wrapped on my computer via your script.
![image](https://user-images.githubusercontent.com/58484289/128460113-d7324f36-cd2c-487a-8dd2-8c862dead782.png)
Perhaps it's my computer's problem. I need to do some further checks.

How about the intention in Makefile block?
I found a way to show the indention in Makefile block:
```diff
--- a/html.cfg
+++ b/html.cfg
@@ -38,7 +38,8 @@
     font-style: italic; 
     color: gray;
     width: 1.5rem;
-    display: inline-block;
+    display: inline-table;
+    white-space: pre;
     -webkit-user-select: none;
     -moz-user-select: none;
     -o-user-select: none;
```

However, the long code still be wrapped and I don't know whether it is caused by `inline-table` or 
it is my computer's problem.
Also I have no idea why `inline-table` worked with `white-space: pre`  but `inline-block` doesn't.

Reply:
The solution of indention problem about Makefile block would be as same as the `*.c` source code. It would be nice to use space instead of tab in the `*.tex` source code.

Let's see the output HTML file converting via [make4ht](https://github.com/michal-h21/make4ht):

![image](https://user-images.githubusercontent.com/26391143/128478270-9688826f-214b-45d2-94c0-70c67248d5aa.png)

The tab couldn't be converted to anything in HTML if we use tab for indentation in `*.tex` file.

Reply:
Here are the further explanations:

In HTML, not only tabs but also spaces could be well rendered in `<pre>` element:

![image](https://user-images.githubusercontent.com/26391143/128482085-58cf17a7-e869-4406-b412-c57cb947a7ea.png)

But the converting tool [make4ht](https://github.com/michal-h21/make4ht) we use didn't put the tab inside the `<span>` of code.

Reply:
For me, replace `inline-block` with `inline-table` and add `white-space: pre` 
rendered tab indention well using github action, and you can check it in [here](https://fennecj.github.io/lkmpg/)

Reply:
這次用中文吧，可能上面寫的不太清楚沒讓你明白。如果你打開瀏覽器的開發者工具，去檢視你生成的那個頁面中的 HTML 情況，會是這樣的：

![image](https://user-images.githubusercontent.com/26391143/128510409-21a10873-8dfd-4197-aa8d-d063191b4431.png)

你會發現，那些在 LaTeX 原始碼中 `make -C /lib/modules/...` 之前使用 TAB 進行縮排的部分，在 make4ht 進行轉換時，他是被放入行號的那個 `<span class="ecrm-0500"> </span>` 中，但對於網頁上的排版來說，最理想的狀況應該是行號部分的 `<span>` 裡面就只放行號，這樣我們才可以讓行號那個 `<span>` 利用固定寬度來調整。

下面是一個更直接的例子，分別是使用 TAB 縮排跟使用空白縮排，轉換成 HTML 內容在反白之後的不同狀況：

| **TAB** | **SPACE** |
| :--: | :--: |
| ![image](https://user-images.githubusercontent.com/26391143/128511696-aa215e2f-d781-4574-8f1f-5b3ec039fe0a.png) | ![image](https://user-images.githubusercontent.com/26391143/128511878-34fbe55b-4875-4400-a5a1-aadb406ef3ae.png) |

Reply:
Thanks for explanation. I agree with that we should use space instead of tab in tex source file.